src/Expr.cc

```diff
@@ -1520,6 +1529,14 @@ AddExpr::AddExpr(ExprPtr arg_op1, ExprPtr arg_op2)
 	else
 		ExprError("requires arithmetic operands");
 
+	// Warn about scalar vector operations after we error about
+	// requiring arithmetic operands.
+	if ( ! IsError() && IsScalarVecOp() )
+		{
+		reporter->Warning("mixing vector and scalar operands is depreciated (%s) (%s)",
+		                  type_name(op1->GetType()->Tag()), type_name(op2->GetType()->Tag()));
+		}
+
```

Comment:
To compress the code here and below a bit, I would suggest factoring this out into a function `CheckScalarVecOp` (or `CheckScalarAggOp`), potentially just combining it with the new `IsScalarVecOp()`.

Reply:
(Also a typo, `depreciated` -> `deprecated`)

Reply:
Thanks for the feedback. Ought to be resolved in the `fixup!` commit. I'll rebase / squash if everything looks good.